sw/device/tests/sim_dv/csrng_fuse_en_sw_app_read.c

```diff
@@ -0,0 +1,225 @@
+
+  check_internal_state(csrng, fail_expected ? &kZeroState : &kExpectedState);
+}
+
+/**
+ * CTR DRBG Known-Answer-Test (KAT) for GENERATE command.
+ */
+static void test_fuse_enable(const dif_csrng_t *csrng) {
+  fips_instantiate_kat(csrng, /*fail_expected=*/false);
+  LOG_INFO("%s", __func__);
+
+  uint32_t got[kExpectedOutputLen];
+
+  csrng_testutils_cmd_generate_run(csrng, got, kExpectedOutputLen);
+  csrng_testutils_cmd_generate_run(csrng, got, kExpectedOutputLen);
+  const dif_csrng_internal_state_t kExpectedState = {
+      .reseed_counter = 3,
+      .v = {0xe73e3392, 0x7d2e92b1, 0x1a0bac9d, 0x53c78ac6},
+
+      .key = {0x66d1b85a, 0xc19d4dfd, 0x053b73e3, 0xe9dc0f90, 0x3f015bc8,
+              0x4436e5fd, 0x1cccc697, 0x1a1c6e5f},
+      .instantiated = true,
+      .fips_compliance = false,
+  };
+  check_internal_state(csrng, &kExpectedState);
+
+  // TODO(#13342): csrng does not provide a linear output order. For example,
+  // note the test vector output word order: 12,13,14,15 8,9,10,11 4,5,6,7
+  // 0,1,2,3.
+  const uint32_t kExpectedOutput[kExpectedOutputLen] = {
+      0xe48bb8cb, 0x1012c84c, 0x5af8a7f1, 0xd1c07cd9, 0xdf82ab22, 0x771c619b,
+      0xd40fccb1, 0x87189e99, 0x510494b3, 0x64f7ac0c, 0x2581f391, 0x80b1dc2f,
+      0x793e01c5, 0x87b107ae, 0xdb17514c, 0xa43c41b7,
+  };
+
+  CHECK_ARRAYS_EQ(got, kExpectedOutput, kExpectedOutputLen,
+                  "Generate command KAT output mismatch");
+}
+
+/**
+ * Check that the internal states will read zero.
+ */
+static void test_fuse_disable(const dif_csrng_t *csrng) {
+  LOG_INFO("%s", __func__);
+  fips_instantiate_kat(csrng, /*fail_expected=*/true);
+}
+
+/**
+ * Read the otp at `HW_CFG.OTP_CTRL_PARAM_EN_CSRNG_SW_APP_READ_OFFSET` address
+ * and check whether is configured by the `uvm_test_seq` as expected.
+ *
+ * @param expected Define the expected value for the
+ * HW_CFG.EN_ENTROPY_SRC_FW_READ flag.
+ */
+static void check_csrng_fuse_enabled(bool expected) {
+  dif_otp_ctrl_t otp;
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp));
+
+  dif_otp_ctrl_config_t config = {
+      .check_timeout = 100000,
+      .integrity_period_mask = 0x3ffff,
+      .consistency_period_mask = 0x3ffffff,
+  };
+  CHECK_DIF_OK(dif_otp_ctrl_configure(&otp, config));
+  otp_ctrl_testutils_wait_for_dai(&otp);
+
+  uint32_t value;
+  // Read the current value of the partition.
+  CHECK_DIF_OK(dif_otp_ctrl_dai_read_start(&otp, kDifOtpCtrlPartitionHwCfg,
+                                           kOtpIfetchHwRelativeOffset));
+  otp_ctrl_testutils_wait_for_dai(&otp);
+  CHECK_DIF_OK(dif_otp_ctrl_dai_read32_end(&otp, &value));
+  multi_bit_bool_t enable = bitfield_field32_read(
+      value,
+      (bitfield_field32_t){.mask = 0xff, .index = kOtpCsrngFwReadBitOffset});
+  CHECK((enable == kMultiBitBool8True) == expected,
+        "`fw_enable` not expected (%x)", enable);
+}
+
+/**
+ * This test:
```

Comment:
Nit: This test -> This test takes the following steps:
(To be consistent with the verbs below.)

Reply:
done